src/App.js

```diff
@@ -1,5 +1,89 @@
+// 클래스 모듈 선언
+const { Console } = require('@woowacourse/mission-utils');
+const Computer = require('./Computer');
+// 상수 선언
+const INPUT_LENGTH = 3;
+
 class App {
-  play() {}
+  #collectValidationFn;
+  #computer;
+
+  constructor() {
+    this.#collectValidationFn = Object.freeze({
+      isNotThreeDigit: inputDigit => inputDigit.length !== INPUT_LENGTH,
+      isNotOneToNineDigit: inputDigit =>
+        isNaN(inputDigit) || inputDigit.toString().includes('0'),
+      isDuplicates: inputDigit => {
+        const arrForCheck = inputDigit.toString().split('');
+        const setForCheck = new Set(arrForCheck);
+        return arrForCheck.length !== setForCheck.size;
+      },
+    });
+  }
+
+  #setUserInput() {
+    Console.readLine('숫자를 입력해주세요 : ', inputDigit => {
+      const userDigit = [...this.isDigitValidation(inputDigit)].map(Number);
+      const baseBallBoard = this.#computer.calcBaseBallDigit(userDigit);
+      this.isThreeStrike(baseBallBoard)
+        ? this.getRestartInput()
+        : this.#setUserInput();
+    });
+  }
+
+  #showStartMessage() {
+    Console.print('숫자 야구 게임을 시작합니다.');
+  }
+
+  #gameStart() {
+    this.#computer = new Computer();
+    this.#setUserInput();
+  }
+
+  #gameEnd() {
+    this.#computer = null;
+    Console.print('게임 종료');
+    Console.close();
+  }
+
+  isThreeStrike(baseBallBoard) {
+    const { strike, ball } = baseBallBoard;
+    if (strike || ball) {
+      Console.print(
+        (ball ? `${ball}볼 ` : ``) + (strike ? `${strike}스트라이크` : ``),
+      );
+    } else Console.print('낫싱');
+    return strike === INPUT_LENGTH;
+  }
+
+  isDigitValidation(inputDigit) {
+    const { isNotThreeDigit, isNotOneToNineDigit, isDuplicates } =
+      this.#collectValidationFn;
+    if (
+      isNotThreeDigit(inputDigit) ||
+      isNotOneToNineDigit(inputDigit) ||
+      isDuplicates(inputDigit)
+    )
```

Comment:
저도 현오님과 같은 방법으로 해결했는데, 현오님 코드보다가 생각난 코드가 있는데요!.. 
 위에서 언급했듯이 이 함수들을 `Validation` 클래스로 빼낸다고 했을 때, 
`isAllPassed()` 라는 메소드를 만들어줘서 아래처럼 만들 수 있겠는걸요?!!🤔
 ```suggestion
    if (this.validator.allPassed(inputDigit))
```

Reply:
예찬님 의견도 가독성에 좀 더 좋은 것 같습니다! 👍
꼼꼼하게 리뷰 남겨주셔서 감사합니다!

예찬님 코드보면서 배우러 들리겠습니다! 😀